swarm/src/lib.rs

```diff
@@ -1542,13 +1622,63 @@ impl NetworkBehaviour for DummyBehaviour {
         }
     }
 
-    fn inject_event(
-        &mut self,
-        _: PeerId,
-        _: ConnectionId,
-        event: <Self::ConnectionHandler as ConnectionHandler>::OutEvent,
-    ) {
-        void::unreachable(event)
+    fn on_event(&mut self, event: InEvent<Self::ConnectionHandler>) {
+        match event {
+            InEvent::ConnectionHandler {
+                peer_id: _,
+                connection: _,
+                event,
+            } => void::unreachable(event),
+            InEvent::ConnectionEstablished {
+                peer_id: _,
+                connection_id: _,
+                endpoint: _,
+                failed_addresses: _,
+                other_established: _,
+            } => {}
```

Comment:
Why match on all of these in here?

Reply:
To have a compile time error when any of them change, forcing us to revisit each implementation. I guess it is a bit overkill for this one. Let me know if you prefer this to be changed.

Reply:
I think it is a good idea but I'd prefer if we do that in a more prominent place. For example, we could have a dedicated `public_api.rs` in `swarm/tests` that concerns itself with such things. I find it a bit odd that this particular impl should take care of that.

Thoughts?